src/Symfony/Component/VarDumper/Cloner/VarCloner.php

```diff
@@ -268,7 +273,6 @@ protected function doClone($var)
             }
 
             $queue[$i] = $vals;
-            unset($indexedArrays[$i]);
```

Comment:
well, not exactly for this reason in this case: it's because PHP allocates arrays as they grow, and never frees the allocated memory when they shrink. I think this article might explain it: http://jpauli.github.io/2016/04/08/hashtables.html